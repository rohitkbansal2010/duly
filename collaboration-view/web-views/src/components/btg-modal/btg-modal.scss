@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.customWidthModal {
  width: 44rem !important;
  max-width: 44rem !important;
}

.customBackDropModal {
  background-color: $default-text !important;
  opacity: 0.15 !important;
}

.btgModal {
  z-index: 1056 !important;

  &Header {
    border-bottom: none !important;

    &IconContainer {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 100%;
      padding-top: 3.4rem;
    }

    &IconWrapper {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 10rem;
      height: 10rem;
      padding: 0.75rem;
      background-color: rgba($dupage-magenta, 0.3);
      border: none;
      border-radius: 100%;

      & img {
        width: 100%;
        height: 100%;
      }
    }
  }

  &Title {
    @include nexa-heavy-highlight;

    color: $default-text;
    font-weight: 900 !important;
    font-size: 1.75rem !important;
    line-height: 2.25rem !important;
    text-align: center;
  }

  &Footer {
    border-top: none !important;

    &ButtonContainer {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 100%;
      padding-top: 0.25rem;
      padding-bottom: 2rem;
    }

    &Button {
      @include nexa-extra-bold-highlight;

      width: 12.625rem !important;
      height: 2.625rem !important;
      margin: 0 !important;
      color: $white !important;
      background-color: $dupage-dark-blue !important;
      border-color: $dupage-dark-blue !important;
      border-radius: 0.5rem !important;
    }
  }
}
