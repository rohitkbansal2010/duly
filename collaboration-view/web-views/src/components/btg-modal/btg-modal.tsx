import { Button, Modal } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { warningCircleMagentaIcon } from '@icons';
import { hideBTGModal } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { AppDispatch } from '@redux/store';

import styles from './btg-modal.scss';

export const BTGModal = () => {
  const dispatch: AppDispatch = useDispatch();

  const isShowBtgModal = useSelector(({ TODAYS_APPOINTMENTS }: RootState) =>
    TODAYS_APPOINTMENTS.isShowBtgModal);

  const closeModal = () =>
    dispatch(hideBTGModal());

  return (
    <Modal
      animation={false}
      show={isShowBtgModal}
      onHide={closeModal}
      size="lg"
      centered
      dialogClassName={styles.customWidthModal}
      backdropClassName={styles.customBackDropModal}
      className={styles.btgModal}
    >
      <Modal.Header className={styles.btgModalHeader}>
        <div className={styles.btgModalHeaderIconContainer}>
          <div
            className={styles.btgModalHeaderIconWrapper}
          >
            <img
              src={warningCircleMagentaIcon}
              alt="restricted patient record icon"
            />
          </div>
        </div>
      </Modal.Header>
      <Modal.Body>
        <Modal.Title className={styles.btgModalTitle}>
          This is a restricted patient record!
        </Modal.Title>
      </Modal.Body>
      <Modal.Footer className={styles.btgModalFooter}>
        <div className={styles.btgModalFooterButtonContainer}>
          <Button
            variant="primary"
            onClick={closeModal}
            className={styles.btgModalFooterButton}
          >
            Close
          </Button>
        </div>
      </Modal.Footer>
    </Modal>
  );
};
