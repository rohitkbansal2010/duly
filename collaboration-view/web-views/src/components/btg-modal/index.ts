export { BTGModal } from './btg-modal';
