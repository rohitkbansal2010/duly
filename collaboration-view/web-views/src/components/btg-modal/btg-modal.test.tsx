import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { BTGModal } from './btg-modal';

const mockDispatch = jest.fn();
jest.mock('react-redux', () =>
  ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn().mockImplementation(() =>
      true),
    useDispatch: () =>
      mockDispatch,
  }));

describe('BTGModal', () => {
  it('should match snapshot', () => {
    expect(toJson(shallow(<BTGModal />))).toMatchSnapshot();
  });
});
