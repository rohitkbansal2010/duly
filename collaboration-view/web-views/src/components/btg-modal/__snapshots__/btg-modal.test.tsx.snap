// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BTGModal should match snapshot 1`] = `
<Modal
  animation={false}
  autoFocus={true}
  backdrop={true}
  backdropClassName="customBackDropModal"
  centered={true}
  className="btgModal"
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  dialogClassName="customWidthModal"
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
  size="lg"
>
  <ModalHeader
    className="btgModalHeader"
    closeButton={false}
    closeLabel="Close"
  >
    <div
      className="btgModalHeaderIconContainer"
    >
      <div
        className="btgModalHeaderIconWrapper"
      >
        <img
          alt="restricted patient record icon"
          src="warning-circle-magenta.svg"
        />
      </div>
    </div>
  </ModalHeader>
  <ModalBody>
    <ModalTitle
      className="btgModalTitle"
    >
      This is a restricted patient record!
    </ModalTitle>
  </ModalBody>
  <ModalFooter
    className="btgModalFooter"
  >
    <div
      className="btgModalFooterButtonContainer"
    >
      <Button
        active={false}
        className="btgModalFooterButton"
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Close
      </Button>
    </div>
  </ModalFooter>
</Modal>
`;
